robottelo/cli/factory.py

```diff
@@ -2531,3 +2536,215 @@ def setup_org_for_a_rh_repo(options=None):
+        u'organization-id': org['id'],
+        u'repository-id': new_repo['id'],
+    })
+    ContentView.publish({'id': cv['id']})
+    cv = ContentView.info({'id': cv['id']})
+    ContentView.version_promote({
+        'id': cv['versions'][0]['id'],
+        'to-lifecycle-environment-id': lce['id'],
+    })
+
+    # Create puppet environment and associate organization and location
+    env = make_environment({
+        'location-ids': loc['id'],
+        'organization-ids': org['id'],
+    })
+
+    # Search for SmartProxy, and associate location
+    puppet_proxy = Proxy.list()[0]
+    Proxy.update({
+        u'id': puppet_proxy['id'],
+        u'location-ids': loc['id'],
+    })
+
+    # Network
+    # Search for existing domain or create new otherwise. Associate org,
+    # location and dns to it
+    _, _, domain_name = settings.server.hostname.partition('.')
+    domain = Domain.list({'search': 'name={0}'.format(domain_name)})
+    if len(domain) == 1:
+        domain = domain[0]
+        Domain.update({
+            'name': domain_name,
+            'location-ids': loc['id'],
+            'organization-ids': org['id'],
+            'dns-id': puppet_proxy['id'],
+        })
+    else:
+        # Create new domain
+        domain = make_domain({
+            'name': domain_name,
+            'location-ids': loc['id'],
+            'organization-ids': org['id'],
+            'dns-id': puppet_proxy['id'],
+        })
+    # Search if subnet is defined with given network. If so, just update its
+    # relevant fields otherwise create new subnet
+    network = settings.vlan_networking.subnet
+    subnet = Subnet.list({'search': 'network={0}'.format(network)})
+    if len(subnet) == 1:
+        subnet = subnet[0]
+        Subnet.update({
+            'name': subnet['name'],
+            'domain-ids': domain['id'],
+            'location-ids': loc['id'],
+            'organization-ids': org['id'],
+            'dhcp-id': puppet_proxy['id'],
+            'dns-id': puppet_proxy['id'],
+            'tftp-id': puppet_proxy['id'],
+        })
+    else:
+        # Create new subnet
+        subnet = make_subnet({
+            'name': gen_string('alpha'),
+            'network': network,
+            'mask': settings.vlan_networking.netmask,
+            'domain-ids': domain['id'],
+            'location-ids': loc['id'],
+            'organization-ids': org['id'],
+            'dhcp-id': puppet_proxy['id'],
+            'dns-id': puppet_proxy['id'],
+            'tftp-id': puppet_proxy['id'],
+        })
+
+    # Search if Libvirt compute-resource already exists. If so, just update its
+    # relevant fields otherwise, create new compute-resource with 'libvirt'
+    # provider.
+    current_libvirt_url = (
+        LIBVIRT_RESOURCE_URL % settings.compute_resources.libvirt_hostname
+    )
+
+    libvirt_resources = [
+        ComputeResource.info({'id': comp_res['id']}) for comp_res
+        in ComputeResource.list()
+    ]
+    comp_resources = [
```

Comment:
Some kind of naming mismatch here. previous variable is called `libvirt_resources` but i can't see any mention of Libvirt in your query, and this variable is just called `comp_resources` although it searches for libvirt resources.